### DISCUSSION TITLE
How do I diagonalize Hamiltonian for spin 3/2

### DISCUSSION THREAD

**Discussion by marcio053** [QuSpin/QuSpin#555] [Q&A]
Hello, I wrote this code to diagonalize the Hamiltonian to spin 3/2, but the output values ​​are incorrect. I would like some advice on how to solve this problem.

from quspin.operators import hamiltonian 
from quspin.basis import spin_basis_1d 
import numpy as np 
from quspin.basis.basis_1d._basis_1d_core import hcp_basis,hcp_ops
from quspin.basis.basis_1d._basis_1d_core import boson_basis, boson_ops
from quspin.basis.basis_1d import spin_basis_1d
import numpy as _np

L=2
J=1
g=0 
h=1 
Omega=4.5 

def drive(t,Omega):
	return np.cos(Omega*t)
drive_args=[Omega]

basis = spin_basis_1d(L=L,S="3/2",pauli=False)
	
		
#print(basis)

x_field=[[g,i] for i in range(L)]
z_field=[[h,i] for i in range(L)]
J_nn=[[J,i,(i+1)%L] for i in range(L-1)] 

static=[["zz",J_nn],["x",x_field],["z",z_field]]
dynamic=[["x",x_field,drive,drive_args]]

H=hamiltonian(static,dynamic,dtype=np.float64,basis=basis,check_symm=False)
#print(H)
#print(H.toarray())
eigenvalues,eigenvectors = H.eigsh(k=8, which='SA')
print(eigenvalues)
eigenvalues,eigenvectors = H.eigsh(k=8, which='LA')
print(eigenvalues)

The output values ​​are:
[-2.25 -2.25 -1.75 -1.75 -1.25 -1.25 -0.75 -0.75]
[-0.25 -0.25  0.25  0.25  1.25  2.75  2.75  5.25]

**Comment by hcut** [comment]
@marcio053 

You are printing the 8 eigenvalues sorted by Smallest Amplitude 'SA', and Largest Amplitude `LA` respectively. That's why they are different.
Just print `eigenvalues,eigenvectors = H.eigh()` `print(eigenvalues)` , to see it by self.

**Reply by marcio053** [reply]
@researcher0 
thanks

**Reply by weinbe58** [reply]
SA stands for smallest algebraic which simply means going from lowest value up 

LA stands for the opposite, largest values to smallest values

There are also other options like 'LM', 'SM' etc.

https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.linalg.eigsh.html#scipy.sparse.linalg.eigsh
